lib/src/core/toastification_config.dart

```diff
@@ -81,22 +82,20 @@ Widget _defaultAnimationBuilderConfig(
 }
 
 /// Default margin builder for [Toastification]
-EdgeInsetsGeometry _defaultMarginBuilder(AlignmentGeometry alignment) {
+EdgeInsetsGeometry _defaultMarginBuilder(BuildContext context, AlignmentGeometry alignment) {
   final y = alignment is Alignment
       ? alignment.y
       : alignment is AlignmentDirectional
           ? alignment.y
           : 0.0;
-  Alignment.topCenter;
+  final kbdBottomMargin = MediaQuery.of(context).viewInsets.bottom;
 
   log('Margin builder called with y: $y');
   log('Margin builder called with alignment: $alignment');
 
   return switch (y) {
-    >= -1 => const EdgeInsets.only(top: 12),
-    >= -0.5 => const EdgeInsets.only(top: 12),
-    <= 0.5 => const EdgeInsets.only(bottom: 12),
-    <= 1 => const EdgeInsets.only(bottom: 12),
+    <= -0.5 => const EdgeInsets.only(top: 12),
+    >= 0.5 => EdgeInsets.only(bottom: 12 + kbdBottomMargin),
```

Comment:
It really works nice when we use Alignment bottom or any alignment which has y more than +0.5 

But if we use Alignment.center we will have wired behavior.

Check out this video

https://github.com/user-attachments/assets/3cb96c4e-2d16-4ecb-a8ba-d7203bcc2819

Reply:
I would say it's better to add the MediaQuery.viewInsets to all alignments

What do you think @MrLightful 

``` Dart 

  return switch (y) {
    <= -0.5 =>
      const EdgeInsets.only(top: 12) + MediaQuery.of(context).viewInsets,
    >= 0.5 => EdgeInsets.only(bottom: 12) + MediaQuery.of(context).viewInsets,
    _ => EdgeInsets.zero + MediaQuery.of(context).viewInsets,
  };
  ```

Reply:
Yeah, works quite nicely and smooth.